services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -2996,6 +2996,10 @@ private SchemaEntry addValueAndSupersetSchemaEntries(
           newSupersetSchemaEntry.getId(),
           newSuperSetWriteComputeSchema.toString());
     }
+    updateStore(
```

Comment:
I guess this is the main changes. Do you have any existing test to cover the changes here?

Reply:
Actually, I am wondering in the incident, is it Server ZK listener failed to catch up schema update? If this is the case, even you tweak the superset schema update order, this will still happen? (As superset schema listener is store repo's listener but schema is another one... 
Or maybe my concern is already covered by some Slack discussion that I am aware of.

Reply:
I don't have a good test in mind to reproduce the race in an integration test deterministically. Since anything we write might still pass even without the fix. Regarding to the zk watch listener issue you are correct that the internal cache updates are still separate callbacks however that should be covered by the forced schema refresh we currently have in our code (when we encounter a new superset schema ID). The problem that this PR is trying to solve is that in some scenarios the superset schema ID IS UPDATED but the corresponding RMD is not available on zk yet so even with an ad-hoc zk update we still can't find the corresponding schema.